c10/kineto_plugin/register_api.cpp

```diff
@@ -0,0 +1,24 @@
+#include "register_api.h"
+
+#ifdef USE_XPU
+#include <c10/kineto_plugin/xpu/XPUActivityProfiler.h>
+#endif
+
+namespace c10 {
+namespace kineto_plugin {
+
+#define REGISTER_KINETO_PLUGIN(TYPE)                          \
+  #ifdef USE_##TYPE                                           \
+  libkineto::api().registerProfilerFactory(                   \
+      []() -> std::unique_ptr<IActivityProfiler> {            \
+        return std::make_unique<TYPE##ActivityProfiler>(); \
+      });                                                     \
+  #endif
+
+// register new kineto backends in kineto plugin
+void registerKinetoPluginProfiler() {
+  REGISTER_KINETO_PLUGIN(XPU);
```

Comment:
It's not a good design. If we are going to provide a common interface to register a kineto backend, it should not depend on device specific code.
Can we have a more general design here? Reference to allocator design?

Reply:
I see. `registerKinetoPluginProfiler` will be called in `kineto_shim.cpp` file as RAII. Refer to allocator design(new it and leak memory intendedly) can solve timing issue.